### PR TITLE
Corto robustness fix for message timeout

### LINKED ISSUE #384: [Bug]: Integration tests for Corto seem flaky
### Bug Description

In rare cases the integration test package fails due to a timeout error in the following test:
test_sommelier.py::TestAuctionFlows::test_sommelier[ethiopia_profiles-SLIM]

The exchange agent asks “What are the flavor profiles of Ethiopian coffee?” over SLIM, but only receive a timeout response (SlimError.SessionError: message='receive timeout waiting for message'). The code then treats the failed response as if it had a body and does response.payload when response is None → Causes 'NoneType' object has no attribute 'payload' and error code 500.

Re-run of the test package usually solves the issue, and the test suite passes.

Here is an example of the erroneous behavior:
https://github.com/agntcy/coffeeAgntcy/actions/runs/22483051385/job/65125784638#step:9:1364

### Expected Behavior

The test suite should run consistently, with no corner- or edge-cases for occasional failures. Ideally timeout should not occur where it is not specifically required for the test case, and the code itself should be robust enough to handle accidental timeouts without failure.

### Affected Version

v1.0.0

### Steps to Reproduce

Since the error is occasional, there is no surefire way to reproduce the issue, but running Corto integration test suite multiple times, especially in slow environments can cause it to happen.

### Checklist

- [x] I have read the [contributing guidelines](/agntcy/coffeeAgntcy/blob/main/CONTRIBUTING.md)
- [x] I have verified this does not duplicate an existing issue

### PR DESCRIPTION
**Pull request by delthazor**
Add timeout error detections, retry and proper handling (along with corresponding tests) to Corto, because timeout errors caused test flakiness.

# Description

Corto integration tests fail due to occasional unexpected timeout errors which usually disappear after re-run. This solution aims to catch such cases, and handle them as good as possible, with minimal impact.
The first time a timeout is observed, the system tries a graceful retry. If the timeout happens again (or the second try fails with another type of error), it probably means a deeper problem that is out of scope for this change. In this case, the new solutions tries to relay as much info about the info towards the higher layers of the application, as possible, trying to avoid reading error messages like they were proper responses, and extracting none existent payloads.

## Issue Link

Fixes #384 

## Type of Change

- [x] Bugfix
- [ ] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](/agntcy/coffeeAgntcy/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: coffeeAGNTCY/coffee_agents/corto/exchange/agent.py [hunk 1]
```diff
@@ -103,36 +109,39 @@ async def a2a_client_send_message(self, prompt: str):
...
         """
-        try:
-            factory = self.factory
-            a2a_topic = A2AProtocol.create_agent_topic(farm_agent_card)
-            transport = factory.create_transport(
-                DEFAULT_MESSAGE_TRANSPORT,
-                endpoint=TRANSPORT_SERVER_ENDPOINT,
-                # SLIM transport requires a routable name (org/namespace/agent) to build the PyName used for request-reply routing
-                name="default/default/exchange"
-            )
```
Comment (pregnor) [inline coffeeAGNTCY/coffee_agents/corto/exchange/agent.py:116]: **Question**: Why did we throw this comment away? We should inline it to the name field's line IMO.

File: coffeeAGNTCY/coffee_agents/corto/exchange/agent.py [hunk 2]
```diff
@@ -103,36 +109,39 @@ async def a2a_client_send_message(self, prompt: str):
...
+            params=MessageSendParams(
+                message=Message(
+                    message_id=str(uuid4()),
+                    role=Role.user,
+                    parts=[Part(TextPart(text=prompt))],
+                )
+            ),
+        )
+
+        def _parse_response(response):
```
Comment (pregnor) [inline coffeeAGNTCY/coffee_agents/corto/exchange/agent.py]: **Question**: Why are we inlining this function nested under the other function instead of promoting it to a top level function?

I would may be thinking about refactoring the control structure of the `a2a_client_send_message()` to be made up of multiple smaller functions instead of mixing levels of detail.

For example we could have separate helpers to 
 
- assemble the request, 
- parse the response
- implement the retry (especially for an exponential back-off that I advised offline)

And that way this higher level function would just be aggregating calls to those while conveying the higher level logic without the nitty gritty details crammed in place.

File: coffeeAGNTCY/coffee_agents/corto/exchange/errors.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    current = exc
+    while current is not None and id(current) not in seen:
+        seen.add(id(current))
+        if isinstance(current, SlimError.SessionError):
+            return True
+        current = getattr(current, "__cause__", None) or getattr(current, "__context__", None)
+    return False
+
+
+def _is_no_payload_error(exc: BaseException) -> bool:
```
Comment (pregnor) [inline coffeeAGNTCY/coffee_agents/corto/exchange/errors.py:57]: **Question**: Could you check for me if this makes the GPT5Pro error a bit more clear? I have seen exceptions on missing payloads with that one and I'm not sure if this has any impact.

It's ok to say no, I'm just curious.

File: coffeeAGNTCY/coffee_agents/corto/tests/unit/exchange/test_agent_timeout_retry.py
Comment (pregnor) [inline coffeeAGNTCY/coffee_agents/corto/tests/unit/exchange/test_agent_timeout_retry.py:1]: **Question**: I might be wrong here, but couldn't this be made more concise with parametrization of the tests? (table testing)
Are the assert conditions so vastly different they can't be made generic?

File: coffeeAGNTCY/coffee_agents/corto/exchange/agent.py [hunk 4]
```diff
@@ -141,7 +150,37 @@ async def a2a_client_send_message(self, prompt: str):
...
+                        "Remote agent did not respond in time (SLIM receive timeout).",
+                        cause=e2,
+                    ) from e2
+                if _is_no_payload_error(e2):
+                    raise RemoteAgentNoResponseError(
+                        "Remote agent returned no response (missing or invalid payload).",
+                        cause=e2,
+                    ) from e2
+                logger.error("A2A send_message failed after retry: %s", e2)
+                raise
```
Comment (pregnor) [inline coffeeAGNTCY/coffee_agents/corto/exchange/agent.py]: **Question**: What do you think about using exponential backoff here, set for like 30s of timeout with a couple retries on a base of 3 for example (starting with the power of 0).

File: coffeeAGNTCY/coffee_agents/corto/exchange/agent.py (hunk 1 above)
Comment (delthazor) [reply coffeeAGNTCY/coffee_agents/corto/exchange/agent.py:116]: Put it back inline.

File: coffeeAGNTCY/coffee_agents/corto/exchange/agent.py (hunk 2 above)
Comment (delthazor) [reply coffeeAGNTCY/coffee_agents/corto/exchange/agent.py]: Extracted appropriate functions to distill the logic.

File: coffeeAGNTCY/coffee_agents/corto/exchange/errors.py (hunk 3 above)
Comment (delthazor) [reply coffeeAGNTCY/coffee_agents/corto/exchange/errors.py:57]: I didn't see any improvements on that, sorry.

File: coffeeAGNTCY/coffee_agents/corto/tests/unit/exchange/test_agent_timeout_retry.py
Comment (delthazor) [reply coffeeAGNTCY/coffee_agents/corto/tests/unit/exchange/test_agent_timeout_retry.py:1]: Changed the test to a more table-like parameterized approach.

File: coffeeAGNTCY/coffee_agents/corto/exchange/agent.py (hunk 4 above)
Comment (delthazor) [reply coffeeAGNTCY/coffee_agents/corto/exchange/agent.py]: Makes sense, so I added an example version of the exponential backoff retry.
